[add] description with action section

NourGweda1 opened the pull request:
-Added more customization options to a specific section cell, including:
Background color
Image animation
Description text
Additional styling attributes (font style, font color)

-We need to customize a certain cell to enhance its appearance and provide a more tailored UI

-Integrated new theme attributes, such as image width/height, animation, and description styling (font style, font color)
-Updated the existing cell to apply default values when no description is provided, ensuring it behaves like the current action cell

NourGweda1 committed:
[add] description with action section

[add] new section view description with action
[add] background color
[add] description style
[add] option to have the image animated

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsDescriptionWithActionView.swift:
```diff
@@ -0,0 +1,8 @@
+//  Created by OS on 08/09/2024.
+//
+
+import Foundation
```
Why is this file empty?

MrRadi6 commented on Sources/TZSettingsView/TZSettingsList/TZSettingsSectionView.swift:
```diff
@@ -44,6 +44,10 @@ struct TZSettingsSectionView: View {
                             case .action(let action):
                                 TZSettingsActionCellView(configuration: cell.cellConfiguration,
                                                           action: action)
+
```
What is the difference between action and descriptionWithAction, I can not notice any difference

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsCellContentView.swift:
```diff
@@ -50,6 +67,6 @@ struct TZSettingsCellContentView: View {
                     .padding(.trailing, 16)
             }
-        }
+        }.background(configuration.theme?.backgroundColor)
```
If the them is nil, what is the expected background color?

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsCellContentView.swift:
```diff
@@ -27,15 +27,32 @@ struct TZSettingsCellContentView: View {
+                /**
+                 The description of the setting cell.
+                 */
+                if !(configuration.description?.isEmpty ?? true) {
```
```suggestion
                if let description = configuration.description, 
                 !description. isEmpty {
```

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsCellContentView.swift:
```diff
@@ -27,15 +27,32 @@ struct TZSettingsCellContentView: View {
              The image to be displayed in the setting cell.
              */
             configuration.image
```
```suggestion
            if let theme = configuration.theme { 
               configuration.image
               .frame(maxWidth: 20, maxHeight: 20)
                .frame(maxWidth: theme.imageWidth, maxHeight: theme.imageHeight)
                .scaleEffect(((theme.isAnimated) ?  theme.scaleEffectValue : 1.0))
                .animation(configuration.animation, value: (theme.isAnimated))
            }
```

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsCellContentView.swift:
```diff
@@ -27,15 +27,32 @@ struct TZSettingsCellContentView: View {
+                .scaleEffect(((configuration.theme?.isAnimated ?? false) ?
+                              configuration.theme?.scaleEffectValue :
+                                1.0) ?? 1.0)
+                .animation(configuration.animation, value: (configuration.theme?.isAnimated ?? false))
```
Is `configuration.theme?.isAnimated` State? when the isAnimated value will change?

MrRadi6 commented on Sources/TZSettingsView/TZSettingsCell/TZSettingsCellConfiguration.swift:
```diff
@@ -65,16 +72,31 @@ extension TZSettingsView {
+        /**
+         The description text to be displayed in the cell, if any.
+         */
+        var description: String?
```
Why is the description optional while the cell type should be a cell with a description?

NourGweda1 replied:
I deleted the cell type description with action since it matches the action cell , so I kept the description this way in case we don't need to use it

NourGweda1 committed:
[edit] is animated value to be binding

[remove] unwanted cell type

NourGweda1 committed:
section background color

- edit background color to be applied on whole section
- edit description on each variable in theme

NourGweda1 committed:
add background color to the main section view

NourGweda1 committed:
remove unwanted code

NourGweda1 committed:
edit background color

NourGweda1 committed:
edit section background without title

NourGweda1 committed:
add vertical padding in section view

NourGweda1 committed:
Merge pull request #3 from TrianglZ/enhancement/section-background

section background
